src/main/java/de/ustutt/iaas/cc/core/RemoteTextProcessorMulti.java

```diff
@@ -13,18 +21,30 @@
  */
 public class RemoteTextProcessorMulti implements ITextProcessor {
 
+	
+	// TODO Insert ConcurrentLinkedQueue for concurrency optimization
+	private ConcurrentLinkedQueue<String> textProcessorRessources;
+	private Client client;
+	private final static Logger logger = LoggerFactory.getLogger(RemoteTextProcessorMulti.class);
+	
 	public RemoteTextProcessorMulti(List<String> textProcessorResources, Client client) {
 		super();
-		// TODO ...
+		this.textProcessorRessources = new ConcurrentLinkedQueue<String>(textProcessorResources);
+		this.client = client;
 	}
 
 	@Override
 	public String process(String text) {
-		// text processing placeholder, to be replaced by your solution :-)
-		String processedText = "[processed by incomplete RemoteTextProcessorMulti] - " + text;
-
-		// TODO send request to "next" text processor endpoint (following some load balancing strategy)
-
+				
+		String ressource = textProcessorRessources.poll();		
+		WebTarget target = client.target(ressource);

```

Comment:
You only need to create the WebTarget objects once (as far as I know), i.e. you can instantiate them in the constructor and then use a list of WebTargets here instead a list of Strings.